### PR TITLE
bpo-43959: clarify the documentation of the PyContextVar C-API

### PR DESCRIPTION
**Pull request by scoder**
https://bugs.python.org/issue43959

Automerge-Triggered-By: GH:scoder

### CODE REVIEW

File: Doc/c-api/contextvars.rst [hunk 1]
```diff
@@ -107,9 +107,9 @@ Context variable functions:
 .. c:function:: PyObject *PyContextVar_New(const char *name, PyObject *def)
 
    Create a new ``ContextVar`` object.  The *name* parameter is used
-   for introspection and debug purposes.  The *def* parameter may optionally
-   specify the default value for the context variable.  If an error has
-   occurred, this function returns ``NULL``.
+   for introspection and debug purposes.  The *def* parameter specifies
+   a default value for the context variable, or ``NULL`` for no default.
```
Comment (orsenthil) [inline Doc/c-api/contextvars.rst:111]: >  or ``NULL`` for no default.

Is this accurate? I don't see this in code.

Comment (scoder) [reply Doc/c-api/contextvars.rst:111]: See? The documentation didn't give that away. :)

Note the little `Py_XINCREF()` here:

https://github.com/python/cpython/blob/bcb094b41f7fe4dd1686c50891d85632fcf0d481/Python/context.c#L829-L830

Comment (orsenthil) [reply Doc/c-api/contextvars.rst:111]: oh yeah! That was hard to recognize.

File: Doc/c-api/contextvars.rst [hunk 2]
```diff
@@ -124,13 +124,12 @@ Context variable functions:
    - the default value of *var*, if not ``NULL``;
    - ``NULL``
 
-   If the value was found, the function will create a new reference to it.
+   Except for ``NULL``, the function returns a new reference.
```
Comment (orsenthil) [inline Doc/c-api/contextvars.rst:127]: Better, this is explicit.

File: Doc/c-api/contextvars.rst [hunk 3]
```diff
@@ -124,13 +124,12 @@ Context variable functions:
...
 
 .. c:function:: PyObject *PyContextVar_Set(PyObject *var, PyObject *value)
 
-   Set the value of *var* to *value* in the current context.  Returns a
-   pointer to a :c:type:`PyObject` object, or ``NULL`` if an error
-   has occurred.
+   Set the value of *var* to *value* in the current context.  Returns
+   a new token object for this change, or ``NULL`` if an error has occurred.
```
Comment (orsenthil) [inline Doc/c-api/contextvars.rst:132]: Should we hyperlink to PyContextToken for the term `token object` ?

**Review by orsenthil — APPROVED** [review]
The changes look good to me.

A hyperlink to PyContextToken will be helpful.

**Pull request merged**